pkg/windows/windows.go

```diff
@@ -152,8 +152,9 @@ type Windows interface {
 	// to the Windows VM if it is not present or if it has the incorrect contents. The remote directory is created if it
 	// does not exist.
 	EnsureFile(*payload.FileInfo, string) error
-	// FileExists returns true if a specific file exists at the given path on the Windows VM
-	FileExists(string) (bool, error)
+	// FileExists returns true if a specific file exists at the given path and checksum on the Windows VM. Set an
+	// empty checksum (checksum == "") to disable checksum check.
+	FileExists(string, string) (bool, error)
```

Comment:
Any reason to add checksum validation here? Can't we have a separate API for that? Function name and what it does is not matching.

Reply:
> Function name and what it does is not matching.

The function name describes the goal and the function arguments the configuration, with `path` and `checksum.


> Can't we have a separate API for that? 

I don't anticipate any use case for it.


> Any reason to add checksum validation here? 

Yes, in this context,  the `checksum` validation is tightly-coupled with the existence of the file in the given `path`, ensuring the expected content is present in the file.